src/mainwindow.cpp

```diff
@@ -921,11 +921,28 @@ void ShowSummary(TreeModel* model, QWidget* parent)
         std::unique_ptr<CounterMap> SubCounters;
     };
 
+    QString beginQueryEventKey = "begin-query";
+    bool foundAQuery = false;
+    for (int i = 0; i < model->rowCount() && !foundAQuery; i++)
+    {
+        QModelIndex valIndex = model->index(i, COL::Value);
+        QString keyString = model->GetEvent(valIndex)["k"].toString();
+        if (keyString == "begin-query")
+        {
+            foundAQuery = true;
```

Comment:
You can `break;` instead of using this variable to control the flow of the loop

Reply:
changed to break;